File: src/DataLoader.jl
module DataLoader
    using CSV
    using DataFrames
    using XLSX
    using MLStyle
    using Revise
    export load_data, check_missing_value

    """
    Description:
    Check the data is exist or not, if not exist, then the path
    probably wrong
    """
    function check_file_exists(file_path::String)
        flag = false
            if isfile(file_path)
                println("File is exists")
                flag = true
            else
            println("File does not exist,check the file path")
        end
        return flag
    end

    function check_file_format(file_path::String)
        _, f_ends = split(file_path, '.')
    return f_ends
    end

    function load_data(file_path::String)
        flag = check_file_exists(file_path)
        if flag
            f_ends = check_file_format(file_path)
            f = @match f_ends begin
                "csv" => "CSV.read(\"$file_path\",DataFrame,normalizenames=true)"
                "txt" => "CSV.read(\"$file_path\",DataFrame,normalizenames=true)"
                "xlsx"=> "XLSX.read(\"$file_path\",DataFrame)"
                "dat"=> "CSV.read(\"$file_path\",DataFrame,normalizenames=true)"
                _ => "not supported yet"
            end
            df = eval(Meta.parse(f))
            return df
        end
    end


    """
    Description:
    Check the missing value of in each column of the dataframe,
    and also print the percentage of the missing value
    """
    function check_missing_value(df::DataFrame)
        l = nrow(df)
        for col in names(df)
            if any(ismissing, df[!, col])
                count_missing = count(ismissing, df[!, col])
                println("Column $col has $count_missing missing values, $(count_missing/l*100)% of the data is missing")
            else
                println("the $col has no missing values!")
            end
        end
    end

end # module DataLoader

File: PyVersion/GPPSimulations/src/data_loader.py
import os
from os.path import split
import pandas as pd


class DataLoader:
    def __init__(self,file_path):
        if not os.path.exists(file_path):
            raise FileNotFoundError(f"{file_path} does not exist")
        else:
            print(f"File found: {file_path}")
            self.file_path = file_path

    def check_data_types(self):
        file_format = self.file_path.split(".")[-1]
        print(f"File format: {file_format}")
        return file_format

    def load_data(self):
        flag = self.check_data_types()
        print(flag)

        if flag == "csv":
            df = pd.read_csv(self.file_path)
        elif flag == "xlsx":
            df = pd.read_excel(self.file_path)
        elif (flag == "dat") | (flag == "txt"):
            df = pd.read_table(self.file_path)
        else:
            raise ValueError("File format not supported")

        return df

    def check_missing(self):
        df = self.load_data()
        missings = df.isnull().sum()
        print(missings)
        return missings

